### ISSUE TITLE
How to do ListItemsSourceProperty on List Item Properties?

### ISSUE THREAD

**Issue by verybadsoldier** [PropertyTools/PropertyTools#202]
I am using the `PropertyGrid` and I want to use the `ItemsSourceProperty` with a list of items.

Let's say I have a list of these items (I will try to explain using the existing `PropertyGrid` example):
```
[Category("Lists|Collection")]
public Collection<Item> Collection { get; set; }
```
This is the (unmodified) `Item` class:
```
public class Item : Observable
    {
        private string name;

        private int number;

        private double fraction;

        public string Name
        {
            get
            {
                return this.name;
            }
            set
            {
                this.SetValue(ref name, value);
            }
        }

        public int Number
        {
            get
            {
                return this.number;
            }
            set
            {
                this.SetValue(ref this.number, value);
            }
        }

        public double Fraction
        {
            get
            {
                return this.fraction;
            }
            set
            {
                this.SetValue(ref this.fraction, value);
            }
        }
    }
```
This is currently displayed as this:
![image](https://user-images.githubusercontent.com/364469/45917457-e3fa0880-be77-11e8-91a4-22db1decceea.png)

But now I want the property `Name` to be a ComboBox offering a list of possible values to click on (`ItemsSource`) like in this example:
![image](https://user-images.githubusercontent.com/364469/45917470-202d6900-be78-11e8-8b5a-af3df2187212.png)

I think I understand how `ListItemItemsSourceProperty` can be used on data that is a simple list of strings (like in the example) but I do not understand how I could possibly use it for a property of an item in a list.

Basically something like doing this (this is just fanatasy code ;):
```
        [Browsable(false)]
        public IEnumerable<string> AvailableNames => new[] { "Hugo", "Carl", "Matt" };

        [Category("Observable|ObservableCollection")]
        [ListItemItemsSourceProperty("AvailableNames", Item.Name)] //here I would like to use the ItemsSource on the property 'Name' of the 'Item' class
        public ObservableCollection<Item> ObservableCollection { get; set; }
```
I really hope I could make myself clear. Is this possible? Big thanks for help in advance!

**Comment by jhonson27** [comment]
Maybe there is another way.
but I solved it by creating a static list of names in the Parent class.
in the item class I would have a getter that returns that list.
and set it as Item source to the name property.

**Comment by verybadsoldier** [comment]
Thanks for the reply, nice solution! But I think the static list is a problem for me since I need to have different `AvailableNames` lists based on the parent object. 
So let's say I have two parent objects (each containing a list of items) and each having a property `AvailableNames` (which differsfor the two parents). So I think I cannot attach this list of available names down at the `Item` itself since the parent object has to provide that to be able to have different lists.

**Comment by jhonson27** [comment]
another option will be to register to the collection changed of the item collection.
and when Action is "Add".
and set the names List  inside the added object.
don't forget to raise property changed after setting the list.

**Comment by verybadsoldier** [comment]
Hm I have another approach I implemented some time ago and recently ported to current version:
https://github.com/verybadsoldier/PropertyTools/tree/feat/itemssource_on_column

With that you can use the `Column` object to specify the items source for the column like this:
```
        public IEnumerable<Column> ItemColumns { get; } = new[]
        {
            new Column("Name", "Name", null,"*", itemSource: nameof(AvailableNames)),
            new Column("Number", "Number"),
            new Column("Fraction", "Fraction"),
        };

        [ColumnsProperty(nameof(ItemColumns))]
        public Collection<Item> Collection 
        {
            get;
            set;
        }
```

What do you think? I was thinking about making a PR out of it.

**Comment by verybadsoldier** [comment]
Implemented and merged by PR https://github.com/PropertyTools/PropertyTools/pull/232
